### ISSUE TITLE
`connect()` cannot be awaited when a wallet name is not specified

### ISSUE THREAD

**Issue by josefleventon** [cosmology-tech/cosmos-kit#268]
## Issue

Cannot await `connect()`. The modal will open, but the flow will continue before a wallet has been selected.

## Workaround

This can be solved by building a custom modal and returning the wallet name.
`connect()` can be awaited when a wallet name is specified, like so:

```js
await wallet.connect('keplr-extension')
```

**Comment by liujun93** [comment]
@josefleventon There is no way we can `await` user selecting a wallet on UI inside `walletRepo.connect()`, but you can detect the wallet status of `walletRepo.current`. It's the current wallet that user has selected. If `current` is `undefined`, means user hasn't selected any wallet yet. If `current.walletStatus` is connecting, connected or errored, you can continue the flow you want.

**Comment by liujun93** [comment]
@josefleventon yes CosmosKit was designed for single chain connection at the beginning. So it does't support multiple chain connection very well. but we're working on it. We're working on hook `useChains`. cc @pyramation @twhy
